.github/scripts/simple-smoke-test.sh

```diff
@@ -1,49 +1,86 @@
 #!/usr/bin/env bash
 
-set -e
-set -o pipefail
+set -o errexit -o nounset -o pipefail
 
 function test_docker_image() {
     local image=$1
     local container_name=$2
     local expected_distribution_type=$3
+    local expected_version=$4
 
-    if [ "$(docker ps --all --quiet --filter name="$container_name")" ]; then
-      echo "Removing existing '$container_name' container"
-      docker container rm --force "$container_name"
+    remove_container_if_exists "${container_name}"
+
+    check_distribution_type "${image}" "${expected_distribution_type}"
+    check_version "${image}" "${expected_version}"
+
+    test_map_read_write "${image}" "${container_name}"
+}
+
+function remove_container_if_exists() {
+    local container_name=$1
+
+    local containers
+    containers=$(docker ps --all --quiet --filter name="${container_name}")
+
+    if [[ -n ${containers} ]]; then
+      echo "Removing existing '${container_name}' container"
+      docker container rm --force "${container_name}"
     fi
+}
+
+function check_distribution_type() {
+    local image=$1
+    local expected_distribution_type=$2
 
-    echo "Checking if $image is EE"
-    if docker run --rm $image bash -c 'compgen -G lib/*enterprise*'; then
+    echo "Checking ${image} distribution type"
+    if docker run --rm "${image}" bash -c 'compgen -G lib/*enterprise*'; then
       echo "EE contents identified"
       distribution_type="ee"
     else
       echo "No EE contents identified - assuming OSS"
       distribution_type="oss"
     fi
 
-    if [[ "$distribution_type" != "$expected_distribution_type" ]]; then
-      echo "Distribution was $distribution_type, not $expected_distribution_type as expected"
+    if [[ "${distribution_type}" != "${expected_distribution_type}" ]]; then
+      echo "Image ${image} should contain ${expected_distribution_type} distribution but ${distribution_type} was detected"
       exit 1
     fi
+}
+
+function check_version() {
+    local image=$1
+    local expected_version=$2
+
+    echo "Checking ${image} version"
```

Comment:
I think it would be good to print what version was detected. 
![image](https://github.com/hazelcast/hazelcast-docker/assets/1242724/62d812bc-18b7-49eb-ab0c-4d64c9966109)

Reply:
c037fde1ea66d9b57c9feaa45b38f1819d2605d5